dot/build_spec_test.go

```diff
@@ -0,0 +1,286 @@
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := BuildFromDB(tt.args.path)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			if tt.want != nil {
+				assert.Equal(t, tt.want.genesis.Name, got.genesis.Name)
+			}
+			// remove files created for tests
+			err = os.RemoveAll(tt.args.path)
+			assert.NoError(t, err)
+		})
+	}
+}
+
+func TestBuildFromGenesis(t *testing.T) {
+	file := genesis.CreateTestGenesisJSONFile(t, false)
+
+	type args struct {
+		path      string
+		authCount int
+	}
+	tests := []struct {
+		name string
+		args args
+		want *BuildSpec
+		err  error
+	}{
+		{
+			name: "invalid file path",
+			args: args{
+				path: "/invalid/path",
+			},
+			err: errors.New("open /invalid/path: no such file or directory"),
+		},
+		{
+			name: "normal conditions",
+			args: args{
+				path: file,
+			},
+			want: &BuildSpec{genesis: &genesis.Genesis{Name: "test"}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := BuildFromGenesis(tt.args.path, tt.args.authCount)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			if tt.want != nil {
+				assert.Equal(t, tt.want.genesis.Name, got.genesis.Name)
+			}
+		})
+	}
+}
+
+func TestBuildSpec_ToJSONRaw(t *testing.T) {
+	tests := []struct {
+		name    string
+		genesis *genesis.Genesis
+		want    string
+		err     error
+	}{
+		{
+			name:    "normal conditions",
+			genesis: &genesis.Genesis{Name: "test"},
+			want: `{
+    "name": "test",
+    "id": "",
+    "chainType": "",
+    "bootNodes": null,
+    "telemetryEndpoints": null,
+    "protocolId": "",
+    "genesis": {},
+    "properties": null,
+    "forkBlocks": null,
+    "badBlocks": null,
+    "consensusEngine": "",
+    "codeSubstitutes": null
+}`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &BuildSpec{
+				genesis: tt.genesis,
+			}
+			got, err := b.ToJSONRaw()
+			assert.ErrorIs(t, err, tt.err)
+			assert.Equal(t, tt.want, string(got))
+		})
+	}
+}
+
+func TestWriteGenesisSpecFile(t *testing.T) {
+	type args struct {
+		data []byte
+		fp   string
+	}
+	tests := []struct {
+		name      string
+		args      args
+		touchFile bool
+	}{
+		{
+			name: "normal conditions",
+			args: args{
+				data: []byte{1},
+				fp:   filepath.Join(t.TempDir(), "test.file"),
+			},
+		},
+		{
+			name: "existing file",
+			args: args{
+				data: []byte{1},
+			},
+			touchFile: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var expectedErr error
+			if tt.touchFile {
+				path := filepath.Join(t.TempDir(), "test.txt")
+				err := os.WriteFile(path, nil, os.ModePerm)
+				require.NoError(t, err)
+				tt.args.fp = path
+				expectedErr = errors.New("file " + path + " already exists, rename to avoid overwriting")
```

Comment:
supernit Have `expectedErr` as `expectedErrMessage` since we only assert the error strings

Reply:
Updated.